### ISSUE TITLE
agent-nodepool Terraform template using an experimental feature without enabling the feature

### ISSUE THREAD

**Issue by Woovie** [rancherfederal/rke2-aws-tf#100]
The `optional()` feature is used here:

https://github.com/rancherfederal/rke2-aws-tf/blob/master/modules/agent-nodepool/variables.tf#L83-L84

To enable the feature, this needs to be included in a Terraform template:

```tf
terraform {
  experiments = [module_variable_optional_attrs]
}
```

This should be shipped in the same directory as variables.tf

**Comment by adamacosta** [comment]
This is actually already fixed on a branch I have open right now to get our examples up-to-date and fully working, but the fix there isn't to enable it as an experimental feature. It's to require Terraform >= 1.3, which is when the feature became non-experimental.

**Comment by adamacosta** [comment]
Fixed by #102
